rust/src/mqtt/mqtt.rs

```diff
@@ -462,14 +462,15 @@ impl MQTTState {
         }
 
         while !current.is_empty() {
-            SCLogDebug!("request: handling {}", current.len());
+            SCLogDebug!("request: handling {}; rem: {}", current.len(), input.len() - current.len());
             match parse_message(current, self.protocol_version, self.max_msg_len) {
                 Ok((rem, msg)) => {
+                    SCLogDebug!("request: handling {}; rem: {}", current.len(), input.len() - current.len());
```

Comment:
Why are you duplicating the log message ?
And why are you logging `input.len() - current.len()` as rem (and not `rem.len()`)